### ISSUE TITLE
`media:delete-unused` Command fails if media is an a11y document reference

### ISSUE THREAD

**Issue by DrauzJu** [shopware/platform#11751]
### Shopware Version

6.6.10.5

### Affected area / extension

Platform(Default)

### Actual behaviour

When running `bin/console media:delete-unused`, it fails with a foreign key error: 

```
[PDOException (23000)]
SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`db`.`document`, CONSTRAINT `fk.document.document_a11y_media_file_id` FOREIGN KEY (`document_a11y_media_file_id`) REFERENCES `media` (`id`) ON DELETE RESTRICT ON UPDATE
```

This is because the MediaDefinition does not have a OneToManyAssociationField for the `document_a11y_media_file_id` of the DocumentDefinition and the UnusedMediaPurger can therefore not detect that the media is actually used.

### Expected behaviour

The media entity is not considered for deletion if it is referenced by an a11y document.

### How to reproduce

Create an a11y document, then run `bin/console media:delete-unused`

### Definition of Done

- [x] An automated test covers the fix.
- [x] Integration/E2E testing in staging is done.
- [ ] Changelog markdown file created or updated in /changelog/_unreleased directory.
- [ ] Developer documentation written/updated.
- [ ] End user and/or developer documentation written or updated with clarification.
- [ ] Regression test created to ensure bug does not reappear, and automated regression tests pass.
- [x] Code changes resolve the reported bug without introducing new issues.
